python/node_classification.py

```diff
@@ -0,0 +1,342 @@
+        out_channels=global_params[ModelParams.OUT_CHANNELS],
+        aggr=global_params[ModelParams.AGGREGATOR]
+    )
+
+    global opt
+    # obtain function by its name from library
+    Optim = getattr(torch.optim, global_params[OptimizerParams.OPTIMIZER])
+    opt = Optim(
+        model.parameters(),
+        lr=global_params[OptimizerParams.LEARNING_RATE],
+        weight_decay=global_params[OptimizerParams.WEIGHT_DECAY])
+
+    global criterion
+    criterion = torch.nn.CrossEntropyLoss()
+
+
+@mgp.read_proc
+def set_model_parameters(
+    ctx: mgp.ProcCtx,
+    params: mgp.Map = {}
+) -> mgp.Record():
+    """The purpose of this function is to initialize all global variables. Parameter 
+    params is used for variables written in query module. It first checks
+    if (new) variables in params are defined appropriately. If so, map of default 
+    global parameters is overriden with user defined dictionary params. 
+    After that it executes previously defined functions declare_globals and 
+    declare_model_and_data and sets each global variable to some value.
+
+    Args:
+        ctx: (mgp.ProcCtx): current context,
+        params: (mgp.Map, optional): user defined parameters from query module. Defaults to {}
+
+    Raises:
+        Exception: exception is raised if some variable in dictionary params is not 
+                    defined as it should be
+
+    Returns:
+        mgp.Record(): empty record
+    """
+    global DEFINED_INPUT_TYPES, DEFAULT_VALUES
+
+    # function checks if input values in dictionary are correctly typed
+    def is_correctly_typed(defined_types, input_values):
+        if isinstance(defined_types, dict) and isinstance(input_values, dict):
+            # defined_types is a dict of types
+            return all(
+                k in input_values  # check if exists
+                and is_correctly_typed(
+                    defined_types[k], input_values[k]
+                )  # check for correct type
+                for k in defined_types
+            )
+        elif isinstance(defined_types, type):
+            return isinstance(input_values, defined_types)
+        else:
+            return False
+
+    # mgconsole bug
+    if "hidden_features_size" in params.keys():
+        params["hidden_features_size"] = list(params["hidden_features_size"])
+    if "metrics" in params.keys():
+        params["metrics"] = list(params["metrics"])
+
+    params = {**DEFAULT_VALUES, **params}  # override any default parameters
+
+    print(params)
+
+    if not is_correctly_typed(DEFINED_INPUT_TYPES, params):
+        raise Exception(
+            f"Input dictionary is not correctly typed. Expected following types {DEFINED_INPUT_TYPES}."
+        )
+
+    declare_globals(params)
+    declare_model_and_data(ctx)
+
+    return mgp.Record()
+
+##############################
+# train
+##############################
+
+
+@mgp.read_proc
+def train(
+    no_epochs: int = 100
+) -> mgp.Record():
```

Comment:
Model should be reinitialized here, because it might have been trained previously and parameters would be non-random.

Reply:
Feature you are trying to forbid is my modus operandi for last week haha. The idea is to train model further if it is needed. User will reinitialize model with ´set_model_parameters´ if wanted.